src/Engine/Combat/BattleInstance.java

```diff
@@ -89,24 +100,33 @@ public BattleSummary calculateBattleResults()
 
   /**
    * Utility struct used to facilitate calculating battle results.
+   * Parameters are public to allow modification in Commander.applyCombatModifiers()
+   * One BattleParams is a single attack from a single unit; any counterattack is a second instance. 
    */
   public static class BattleParams
   {
+    public final Unit attacker, defender; 
+    public final BattleInstance combatRef; // strictly for reference
     public double baseDamage;
     public double attackerHP;
     public double attackFactor;
     public double defenderHP;
     public double defenseFactor;
     public double terrainDefense;
+    public final boolean isCounter;
 
-    public BattleParams(Unit attacker, Weapon attackerWeapon, Unit defender, Environment battleground)
+    public BattleParams(Unit attacker, Weapon attackerWeapon, Unit defender, Environment battleground, boolean isCounter, final BattleInstance ref)
     {
+      this.attacker = attacker;
+      this.defender = defender;
       baseDamage = attackerWeapon.getDamage(defender.model);
       attackFactor = attacker.model.getDamageRatio();
       attackerHP = attacker.getHP();
       defenseFactor = defender.model.getDefenseRatio();
       defenderHP = defender.getHP();
       terrainDefense = 0;
+      this.isCounter = isCounter;
+      combatRef = ref;
```

Comment:
Is there any need to store `combatRef`, or is this just an "in case it's useful one day" kind of thing?

Reply:
Could always add `range` to params as final. Or just figure it out based on unit positions. There shouldn't be a need to change that even in `BattleInstance`, I'd think. The range is what it is.
The Commander shouldn't need to get to the map via `combatRef` either--and probably shouldn't be allowed to, since can't this be a `MapMaster`? He's got his own `MapWindow` locally if he wants to look around.

Reply:
Can't figure it out based on unit positions, necessarily, since the attacker is at the start of his move path.
Could use the combatRef's attackerX/Y, though.
It can be a `MapMaster`, yes. The idea is that you might have a CO that needs map info to make proper damage calculations, and that should be based on full map info.

Reply:
So in short, yes - it is an "in case it's useful one day" thing. I'm not sure I like it though, from a design/data encapsulation perspective. I also shudder a bit whenever a private variable changes to public (like everything in BattleInstance, for example). This is just more of the flexibility vs correctness discussion, and how much time we want to spend on getting it "right". I like to lean towards clean implementations when possible, but I also like to see results in my lifetime.